components/de_amp/browser/de_amp_url_loader.cc

```diff
@@ -0,0 +1,404 @@
+/* Copyright 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/de_amp/browser/de_amp_url_loader.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <iostream>
+#include <regex>
```

Comment:
you could try using `"third_party/re2/src/re2/re2.h"`.
For simple regexps there is `base/strings/pattern.h`